### PR TITLE
Bluetooth: Mesh: Don't reset NULL buffer in DM srv

### PR DESCRIPTION
**Pull request by ludvigsj**
This removes a reset of a NULL buffer in the DM server model reset handler.

Signed-off-by: Ludvig Samuelsen Jordet <ludvig.jordet@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/vnd/dm_srv.c [hunk 1]
```diff
@@ -458,7 +458,6 @@ static void bt_mesh_dm_srv_reset(struct bt_mesh_model *model)
 	srv->cfg.ttl = CONFIG_BT_MESH_DM_SRV_DEFAULT_TTL;
 	srv->cfg.timeout = CONFIG_BT_MESH_DM_SRV_DEFAULT_TIMEOUT,
 	srv->cfg.delay = CONFIG_BT_MESH_DM_SRV_REFLECTOR_DELAY,
-	net_buf_simple_reset(srv->pub.msg);
```
Comment (PavelVPV) [inline subsys/bluetooth/mesh/vnd/dm_srv.c:461]: Do we need the publication context then?
https://github.com/nrfconnect/sdk-nrf/blob/c24fe6fbccfdcb47f7223740b3f8078f62bbff7e/include/bluetooth/mesh/vnd/dm_srv.h#L33-L34

**Review by Balaklaka — COMMENTED** [review]
I agree with @PavelVPV

File: subsys/bluetooth/mesh/vnd/dm_srv.c (hunk 1 above)
Comment (ludvigsj) [reply subsys/bluetooth/mesh/vnd/dm_srv.c:461]: Yep, as far as I can read from the docs, it is required in the BT_MESH_MODEL_VND_CB macro:
[link to zephyr docs](https://docs.zephyrproject.org/apidoc/latest/group__bt__mesh__access.html#gabe79d914990ba2721f5bb7081910e512)

**Review by Balaklaka — APPROVED** [review]

File: subsys/bluetooth/mesh/vnd/dm_srv.c (hunk 1 above)
Comment (PavelVPV) [reply subsys/bluetooth/mesh/vnd/dm_srv.c:461]: AFAIR it can be NULL.

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**